src/app/carbon-estimation/carbon-estimation.component.ts

```diff
@@ -149,12 +149,23 @@ export class CarbonEstimationComponent implements OnInit, OnDestroy {
 
   private calculateChartHeight(innerHeight: number, innerWidth: number, expansionPanelHeight: number) {
     // medium tailwind responsive design breakpoint https://tailwindcss.com/docs/responsive-design
-    if (innerWidth < 768) {
-      return innerHeight - this.estimatorBaseHeight - expansionPanelHeight + estimatorHeights.title;
-    }
+    const responsiveBreakpoint = 768;
+
     const extraHeightString = this.extraHeight();
     const extraHeight = Number(extraHeightString) || 0;
-    return innerHeight - this.estimatorBaseHeight - extraHeight - expansionPanelHeight;
+
+    const calculatedHeight =
+      innerWidth < responsiveBreakpoint ?
+        innerHeight - this.estimatorBaseHeight - expansionPanelHeight + estimatorHeights.title
+      : innerHeight - this.estimatorBaseHeight - extraHeight - expansionPanelHeight;
+
+    // Cap on the mininum height of the chart to prevent the chart becoming squashed when zooming
+    // in on desktop browsers. (Zooming in results in window.innerHeight decreasing proportionally
+    // on most desktop browsers which can result in the calculatedHeight above becoming too small
+    // or even negative. N.B. Mobile browsers behave differently.)
+    const minChartHeight = 300;
```

Comment:
I was having some issues related to differences in behaviour between desktop and mobile browsers when trying to calculate a dynamic lower bound or rescaling the chart with zoom so I went with a fixed lower bound.

Reply:
Given we limit the maximum height of the chart in `getChartHeight()`, it feels like it would be cleaner to do both clamping actions in there.